tests/semver/test_sem_ver_utils.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+
+from poetry.semver import is_sem_ver_constraint
+from poetry.semver import sem_ver_sorted
+
+
+@pytest.mark.parametrize(
+    "constraint,result",
```

Comment:
should these also be handled? 

https://legacy.python.org/dev/peps/pep-0440/#local-version-identifiers

Reply:
- punt to backlog/requirement for parse_constraint
- there are more details on those specifics in tests/semver/test_main.py
- a simpler test on this utility is just to mock/spy `parse_single_contraint` to check that it is called with given arguments and leave all the details to the tests on that function; would that be better?

Reply:
- added doc strings
- added mock-spy on parser
- done with amended commits (for now)
- 🎾 > @brycedrennan